roles/prometheus/tasks/main.yml

```diff
@@ -0,0 +1,88 @@
+---
+
+# Copyright 2023 Cloudera, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+- name: Create Prometheus directory 
+  ansible.builtin.file:
+    path: "{{ prometheus_directory }}"
+    state: directory
+
+- name: Create a temporary directory
+  ansible.builtin.tempfile:
+    state: directory
+  register: __prometheus_tmp
+
+- name: Download Prometheus tarball
+  ansible.builtin.get_url:
+    url: "{{ prometheus_tarball_url }}"
+    dest: "{{ __prometheus_tmp.path }}/{{ prometheus_tarball_file }}"
+
+- name: Extract tarball
+  ansible.builtin.unarchive:
+    src: "{{ __prometheus_tmp.path }}/{{ prometheus_tarball_file }}"  
+    dest: "{{ prometheus_directory }}"
+    extra_opts: --strip-components=1
+    remote_src: yes
+
+- name: Remove the temporary  directory 
+  when: __prometheus_tmp is defined
+  ansible.builtin.file:
+    path: "{{ __prometheus_tmp.path }}"
+    state: absent
+
+- name: Create Prometheus user
```

Comment:
This user can be created as a system user (`system: True`). Ref: https://docs.ansible.com/ansible/latest/collections/ansible/builtin/user_module.html#parameter-system

Reply:
Added